src/dune_rules/lib.ml

```diff
@@ -431,9 +427,9 @@ let wrapped t =
       assert false (* will always be specified in dune package *)
     | Some (This x) -> Some x)
 
-let equal l1 l2 = Ordering.is_eq (compare l1 l2)
+let equal = ( == )
 
-let hash t = Id.hash t.unique_id
+let hash = Poly.hash
```

Comment:
Could you remind me how polymorphic hashing works?

Reply:
In this case, we're hashing a record so it works by recursively traversing the record's fields and accumulating a hash as it goes along. Is that what you mean?

Reply:
Yes, thank you.